swan/scripts/03_userconfig.sh

```diff
@@ -0,0 +1,122 @@
+#  if [[ $CONNECTOR_BUNDLED_CONFIGS ]]
+#   then
+#     ln -s $CONNECTOR_BUNDLED_CONFIGS/bundles.json $JUPYTER_CONFIG_DIR/nbconfig/sparkconnector_bundles.json
+#     ln -s $CONNECTOR_BUNDLED_CONFIGS/spark_options.json $JUPYTER_CONFIG_DIR/nbconfig/sparkconnector_spark_options.json
+#   fi
+#  _log "Completed Spark Configuration"
+
+#  SETUP_SPARK_TIME_SEC=$(echo $(date +%s.%N --date="$START_TIME_SETUP_SPARK seconds ago") | bc)
+#  _log "user: $USER, host: ${SERVER_HOSTNAME%%.*}, metric: configure_user_env_spark.${ROOT_LCG_VIEW_NAME:-none}.${SPARK_CLUSTER_NAME:-none}.duration_sec, value: $SETUP_SPARK_TIME_SEC"
+# fi
+
+# Run user startup script
+export JUPYTER_DATA_DIR=$LCG_VIEW/share/jupyter 
+TMP_SCRIPT=`mktemp`
+
+if [[ $USER_ENV_SCRIPT && -f `eval echo $USER_ENV_SCRIPT` ]];
+then
+  START_TIME_SETUP_SCRIPT=$( date +%s.%N )
+
+  _log "Found user script: $USER_ENV_SCRIPT"
+  TMP_SCRIPT=`mktemp`
+  cat `eval echo $USER_ENV_SCRIPT` > $TMP_SCRIPT
+  source $TMP_SCRIPT
+
+  SETUP_SCRIPT_TIME_SEC=$(echo $(date +%s.%N --date="$START_TIME_SETUP_SCRIPT seconds ago") | bc)
+  _log "user: $USER, host: ${SERVER_HOSTNAME%%.*}, metric: configure_user_env_script.duration_sec, value: $SETUP_SCRIPT_TIME_SEC"
+else
+ _log "Cannot find user script: $USER_ENV_SCRIPT";
+fi
+
+# In k8s, $KRB5CCNAME_NB_TERM points to the location of the EOS kerberos ticket that notebook and terminal
+# processes use. The Jupyter server uses the same ticket stored in another location ($KRB5CCNAME). With the
+# code below, we force that $KRB5CCNAME becomes $KRB5CCNAME_NB_TERM only for notebook and terminal processes.
+# This duality exists to prevent the user from overwriting the EOS kerberos ticket that the Jupyter server
+# uses, e.g. by doing `kinit` from a notebook or a terminal.
+# In puppet, $KRB5CCNAME_NB_TERM does not exist. Thus we give it a value and make $KRB5CCNAME point to that
+# value.
+# Both in k8s and puppet, renewals of the EOS kerberos tickets are automatically done to prevent expiration,
+# both for the Jupyter server and the notebooks and terminals. However, if the user runs an explicit kinit
+# and generates new kerberos credentials for notebooks and terminals, they are responsible for renewing those
+# credentials from that point on.
+if [[ -z $KRB5CCNAME_NB_TERM ]]
+then
+  KRB5CCNAME_NB_TERM="/tmp/krb5cc_${USER_ID}_${RANDOM}"
+fi
+export KRB5CCNAME=$KRB5CCNAME_NB_TERM
+
+# As the LCG setup might set PYTHONHOME, run python with -I (Isolated Mode) to prevent
+# the lookup for modules in a Python 3 path and user site
+python -I /srv/singleuser/configure_kernels_and_terminal.py
+
+# Remove our extra paths (where we install our extensions) in the kernel (via SwanKernelEnv kernel extension), 
+# leaving the user env cleaned. It should be the last one called to allow the kernel to load our extensions correctly.
+#  echo "c.InteractiveShellApp.extensions.append('swankernelenv')" >>  $KERNEL_PROFILEPATH
```

Comment:
This extra path is defined in this image, so this should also be added here, no?

Reply:
export KERNEL_PROFILEPATH first and then echo.